app/src/test/java/org/simple/clinic/overdue/AppointmentSyncTest.kt

```diff
@@ -0,0 +1,79 @@
+package org.simple.clinic.overdue
+
+import com.f2prateek.rx.preferences2.Preference
+import com.nhaarman.mockito_kotlin.any
+import com.nhaarman.mockito_kotlin.eq
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.verify
+import com.nhaarman.mockito_kotlin.whenever
+import io.reactivex.Completable
+import io.reactivex.Single
+import junitparams.JUnitParamsRunner
+import junitparams.Parameters
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.simple.clinic.sync.DataPullResponse
+import org.simple.clinic.sync.DataPushResponse
+import org.simple.clinic.sync.SyncCoordinator
+import org.simple.clinic.util.Optional
+
+@Suppress("UNCHECKED_CAST")
+@RunWith(JUnitParamsRunner::class)
+class AppointmentSyncTest {
+
+  private val syncCoordinator = mock<SyncCoordinator>()
+  private val repository = mock<AppointmentRepository>()
+  private val apiV1 = mock<AppointmentSyncApiV1>()
+  private val apiV2 = mock<AppointmentSyncApiV2>()
+  private val lastPullToken = mock<Preference<Optional<String>>>()
+
+  private lateinit var config: AppointmentConfig
+
+  private lateinit var sync: AppointmentSync
+
+  @Before
+  fun setup() {
+    sync = AppointmentSync(syncCoordinator, repository, apiV1, apiV2, Single.fromCallable { config }, lastPullToken)
+  }
+
+  @Test
+  @Parameters("true", "false")
+  fun `when pulling appointments, v2 API should only be called if its enabled`(v2ApiEnabled: Boolean) {
+    whenever(syncCoordinator.pull(eq(repository), eq(lastPullToken), any())).thenAnswer { invocation ->
+      val pullNetworkCall = invocation.arguments[2] as (Int, String?) -> Single<out DataPullResponse<Appointment>>
+      pullNetworkCall.invoke(10, "lastPullToken")
+      Completable.complete()
+    }
+
+    config = AppointmentConfig(v2ApiEnabled = v2ApiEnabled, highlightHighRiskPatients = true)
+
+    sync.pull().blockingAwait()
+
+    if (v2ApiEnabled) {
+      verify(apiV2).pull(any(), any())
```

Comment:
Do you feel we should actually test the values being sent to the methods here instead of making them `any()`?

Reply:
No, the values are already being tested using `AppintmentSyncAndroidTest`. I added these tests only to check that the correct API is used.